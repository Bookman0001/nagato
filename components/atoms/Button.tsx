import styled from 'styled-components'

interface Props {
  children: React.ReactNode
  onClick: () => void
}

export default function Button({ children, onClick }: Props) {
  return (
    <ButtonItem onClick={onClick}>
      <>{children}</>
    </ButtonItem>
  )
}

const ButtonItem = styled.button`
  background-color: #ffffff;
  border-radius: 10px;
  border: 2px solid #303030;
  padding: 5px 3rem;
  display: block;
  margin: 0 auto;
  font-size: 32px;
  cursor: pointer;
`
