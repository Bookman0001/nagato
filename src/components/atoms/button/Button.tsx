import React, { ButtonHTMLAttributes, MouseEvent } from 'react'
import styled from 'styled-components'

import type { ButtonBgColor } from 'src/types/theme'

import { BORDER_RADIUS, COLOR, FONT_SIZE } from 'src/theme/constants'

interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode
  bgColor?: ButtonBgColor
  onClick?: () => void
}

export function Button({
  children,
  bgColor = COLOR.BLACK,
  onClick,
  ...props
}: Props) {
  const handleClick = (e: MouseEvent) => {
    if (onClick) {
      e.stopPropagation()
      e.preventDefault()
      onClick()
    }
  }
  return (
    <ButtonItem onClick={handleClick} bgColor={bgColor} {...props}>
      {children}
    </ButtonItem>
  )
}

const ButtonItem = styled.button<{ bgColor: ButtonBgColor }>`
  min-height: 40px;
  height: 100%;
  border-radius: ${BORDER_RADIUS.DEFAULT};
  padding: 0 1rem;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: ${FONT_SIZE.M};
  background-color: ${({ bgColor }) => bgColor};
  border: 2px solid ${COLOR.WHITE};
  cursor: pointer;
  :hover {
    opacity: 0.9;
  }
  :disabled {
    opacity: 0.7;
  }
`
