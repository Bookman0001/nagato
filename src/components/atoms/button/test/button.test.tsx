import { shallow } from 'enzyme'

import Button from '../'

test('Button', () => {
  const component = shallow(
    <Button onClick={() => console.log('clicked')} text={'test'} />
  )
  expect(component).toMatchSnapshot()
})
