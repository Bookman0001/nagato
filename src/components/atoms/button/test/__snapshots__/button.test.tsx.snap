// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<styled.button
  onClick={[Function]}
>
  test
</styled.button>
`;
